File: csharp/Main/FriendBuilder.cs
using System;

namespace Soat.CleanCoders.DipKata.Main
{
    public class FriendBuilder
    {
        private string   _firstName   = "John";
        private string   _lastName    = "Doe";
        private DateTime _dateOfBirth = new DateTime(1980, 9, 10);
        private string   _email       = "john.doe@foobar.com";

        public static FriendBuilder AFriend()
        {
            return new FriendBuilder();
        }

        public FriendBuilder WithEmail(string email)
        {
            _email = email;
            return this;
        }

        public FriendBuilder WithFirstName(string firstName)
        {
            _firstName = firstName;
            return this;
        }

        public FriendBuilder WithLastName(string lastName)
        {
            _lastName = lastName;
            return this;
        }

        public FriendBuilder WithDateOfBirth(DateTime dateOfBirth)
        {
            _dateOfBirth = dateOfBirth;
            return this;
        }

        public Friend Build()
        {
            return new Friend(_firstName, _lastName, _dateOfBirth, _email);
        }
    }
}

File: java/src/test/java/fr/soat/cleancoders/FriendBuilder.java
package fr.soat.cleancoders;

import fr.soat.cleancoders.highlevel.BirthdayGreeter;

import java.time.LocalDate;

public class FriendBuilder {
    private String firstName = "John";
    private String lastName = "Doe";
    private LocalDate dateOfBirth = LocalDate.of(1980, 9, 10);
    private String email = "john.doe@foobar.com";
    private String phoneNumber = "06xxxxxx";

    public static FriendBuilder aFriend() {
        return new FriendBuilder();
    }

    public FriendBuilder withEmail(String email) {
        this.email = email;
        return this;
    }

    public FriendBuilder withFirstName(String firstName) {
        this.firstName = firstName;
        return this;
    }


    public FriendBuilder withLastName(String lastName) {
        this.lastName = lastName;
        return this;
    }

    public FriendBuilder withDateOfBirth(LocalDate dateOfBirth) {
        this.dateOfBirth = dateOfBirth;
        return this;
    }

    public BirthdayGreeter.Friend build() {
        return new BirthdayGreeter.Friend(firstName, email, phoneNumber);
    }
}
